source: https://mail.gnome.org/archives/gamin-list/2016-April/msg00000.html

The musl C library does not provide the non portable
PTHREAD_MUTEX_RECURSIVE_NP. Test for PTHREAD_MUTEX_RECURSIVE_NP only.

--- a/libgamin/gam_data.c
+++ b/libgamin/gam_data.c
@@ -470,7 +470,7 @@ gamin_data_new(void)
     }
     if (is_threaded > 0) {
        pthread_mutexattr_init(&attr);
-#if defined(linux) || defined(PTHREAD_MUTEX_RECURSIVE_NP)
+#if defined(PTHREAD_MUTEX_RECURSIVE_NP)
        pthread_mutexattr_settype(&attr, PTHREAD_MUTEX_RECURSIVE_NP);
 #else
        pthread_mutexattr_settype(&attr, PTHREAD_MUTEX_RECURSIVE);
